build/gen_stub.php

```diff
@@ -2440,7 +2440,131 @@ function generateClassSynopses(array $classMap): array {
+            continue;
+        }
+
+        $docComparator = new DOMDocument();
+        $docComparator->preserveWhiteSpace = false;
+        $docComparator->formatOutput = true;
+
+        $classSynopsisElements = [];
+        foreach ($doc->getElementsByTagName("classsynopsis") as $element) {
+            $classSynopsisElements[] = $element;
+        }
+
+        foreach ($classSynopsisElements as $classSynopsis) {
+            if (!$classSynopsis instanceof DOMElement) {
+                continue;
+            }
+
+            $firstChild = $classSynopsis->firstElementChild;
+            if ($firstChild === null) {
+                continue;
+            }
+            $firstChild = $firstChild->firstElementChild;
+            if ($firstChild === null) {
+                continue;
+            }
+            $className = $firstChild->textContent;
+            if (!isset($classMap[$className])) {
+                continue;
+            }
+            $classInfo = $classMap[$className];
+
+            $newClassSynopsis = $classInfo->getClassSynopsisElement($doc, $classMap);
+            if ($newClassSynopsis === null) {
+                continue;
+            }
+
+            // Check if there is any change - short circuit if there is not any.
+
+            $xml1 = $doc->saveXML($classSynopsis);
+            $xml1 = getReplacedClassSynopsisXml($xml1);
+            $docComparator->loadXML($xml1);
+            $xml1 = $docComparator->saveXML();
+
+            $classSynopsis->parentNode->replaceChild($newClassSynopsis, $classSynopsis);
+
+            $xml2 = $doc->saveXML($newClassSynopsis);
+            $xml2 = getReplacedClassSynopsisXml($xml2);
+
+            $docComparator->loadXML($xml2);
+            $xml2 = $docComparator->saveXML();
+
+            if ($xml1 === $xml2) {
+                continue;
+            }
+
+            // Return the updated XML
+
+            $replacedXml = $doc->saveXML();
+
+            $replacedXml = preg_replace(
+                [
+                    "/REPLACED-ENTITY-([A-Za-z0-9._{}%-]+?;)/",
+                    "/<phpdoc:classref\s+xmlns:phpdoc=\"([a-z0-9.:\/]+)\"\s+xmlns=\"([a-z0-9.:\/]+)\"\s+xmlns:xi=\"([a-z0-9.:\/]+)\"\s+xml:id=\"([a-z0-9._-]+)\"\s*>/i",
+                    "/<phpdoc:classref\s+xmlns:phpdoc=\"([a-z0-9.:\/]+)\"\s+xmlns=\"([a-z0-9.:\/]+)\"\s+xmlns:xlink=\"([a-z0-9.:\/]+)\"\s+xmlns:xi=\"([a-z0-9.:\/]+)\"\s+xml:id=\"([a-z0-9._-]+)\"\s*>/i",
+                ],
+                [
+                    "&$1",
+                    "<phpdoc:classref xml:id=\"$4\" xmlns:phpdoc=\"$1\" xmlns=\"$2\" xmlns:xi=\"$4\">",
+                    "<phpdoc:classref xml:id=\"$5\" xmlns:phpdoc=\"$1\" xmlns=\"$2\" xmlns:xlink=\"$3\" xmlns:xi=\"$4\">",
+                ],
+                $replacedXml
+            );
+
+            $classSynopses[$pathName] = $replacedXml;
+        }
+    }
+
+    return $classSynopses;
+}
+
+function getReplacedClassSynopsisXml(string $xml): string|array|null
```

Comment:
```suggestion
function getReplacedClassSynopsisXml(string $xml): string
```

Reply:
Oops, I probably forgot that this syntax requires PHP 8. :) (and that `?array` is not even a useful return type)